packages/medusa/src/services/totals.ts

```diff
@@ -2,31 +2,55 @@ import _ from "lodash"
 import { BaseService } from "medusa-interfaces"
 import { MedusaError } from "medusa-core-utils"
 
+import { TaxLine } from "../models/tax-line"
+import { Order } from "../models/order"
+import { Cart } from "../models/cart"
+import { LineItem } from "../models/line-item"
+import { Discount } from "../models/discount"
+import { DiscountRuleType } from "../models/discount-rule"
+
+import TaxProviderService from "./tax-provider"
+import { ITaxCalculationStrategy } from "../interfaces/tax-calculation-strategy"
+import { TaxCalculationContext } from "../interfaces/tax-service"
+
+import {
+  SubtotalOptions,
+  LineDiscount,
+  LineAllocationsMap,
+  LineDiscountAmount,
+} from "../types/totals"
+
 /**
  * A service that calculates total and subtotals for orders, carts etc..
  * @implements {BaseService}
  */
 class TotalsService extends BaseService {
-  constructor() {
+  private taxProviderService_: TaxProviderService
+  private taxCalculationStrategy_: ITaxCalculationStrategy
+
+  constructor({ taxProviderService, taxCalculationStrategy }) {
```

Comment:
nit: this should probably be extracted to something like: 
`type TotalsServiceProps = { ... } `